### PR TITLE
Add support for editorconfig

### PR DESCRIPTION
**Pull request by nojaf**
Current attempt for solving #650.
EditorConfig.Core was forked to https://github.com/nojaf/editorconfig-core-net.
There I created a slightly modified version, mainly having a netstandard2.0 only configuration.

I used ILRepack to merge in the editorconfig core to Fantomas.dll.
I published `Fantomas.EditorConfig.Core` as unlisted to Nuget.

I fear that this PR is already competing with #939, for that @josefblaha I'm sorry.
Last Friday I already had those changes to `readConfiguration `.

Main things left to do here are:
- [x] update unit tests
- [x] integrate into CLI tool
- [x] remove all traces of JSON config (including schema)

### CODE REVIEW

**Comment by josefblaha** [comment]
What an exciting world is this OSS development, with developers racing to complete features ;-)

@nojaf, you are on a good track to finish this, so I will no purse it anymore. Just two things came to my mind when reviewing your code:

1. For ``IndentSpaceNum`` option, there is no one-to-one mapping with editorconfig. IMO the value should be derived from ``indent_style``, ``indent_size``, and ``tab_width`` properties.
2. No warnings should be generated from unrecognized properties; any other custom properties or possible future global properties should be ignored.

**Comment by nojaf** [comment]
Hey @josefblaha, again apologies for not creating my PR sooner. I should have made a draft PR last Friday, that may have avoided this.

As for your feedback:
1. F# doesn't allow tabs so we only need to take spaces into account.
2. Good point, warnings should not be listed anymore as they count have meaning to other things.

**Comment by josefblaha** [comment]
No problem. Any code written has a value for me, as I'm still learning the language.

Ad tabs. You're right, but there is a corner case: ``indent_size = tab`` means that you need to look on ``tab_width`` to get the value. See (https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#widely-supported-by-editors).

**Comment by nojaf** [comment]
Actually, the underlying EditorConfig.Core Nuget solves this internally already,
If I had for example:
```ini
[*.fs]
indent_style=tab
indent_size=tab
tab_width=5
fsharp_indent_on_try_with=true
```

I got
image

I'll add a unit test to cover this as well.

**Comment by nojaf** [comment]
This is ready for now.
A couple of significant changes:
* You don't need to pass down the configuration file in the cmd tool. Fantomas will pick it up based on the original file you passed.
  One trade-off here is that you can't pass a configuration anymore when you use the stdIn. I would add this back in if requested.
  I'm not sure if many people actually use this.
* Same remark for using the FAKE Helpers, config will be picked up based on the F# file, to me this is the way to go.
  I'll update the sample once there is a new alpha released.
* I've renamed IndentSpaceNum and PageWidth to match the .editorconfig names.
* Fantomas-tools would need an update after this PR.

File: src/Fantomas/CodeFormatter.fsi [hunk 1]
```diff
@@ -36,4 +36,4 @@ type CodeFormatter =
 
     /// Accepts a file or a folder and parses the found json to a FormatConfig
```
Comment (jindraivanek) [inline src/Fantomas/CodeFormatter.fsi:37]:
```suggestion
    /// Accepts a file or a folder and parses the found .editorconfig file to a FormatConfig
```

**Pull request merged**